crates/sncast/tests/e2e/declare.rs

```diff
@@ -286,12 +288,75 @@ async fn test_many_packages_default() {
         "99999999999999999",
     ];
 
-    // todo: (1247) Execute tests in temp dirs
     let snapbox = Command::new(cargo_bin!("sncast"))
-        .current_dir(CONTRACTS_DIR.to_string() + "/multiple_packages")
+        .current_dir(tempdir.path())
         .args(args);
     snapbox.assert().failure().stderr_matches(indoc! {r"
         ...
         Error: More than one package found in metadata - specify package using --package flag
     "});
 }
+
+#[tokio::test]
+async fn test_worskpaces_package_specified_virtual_fibonacci() {
+    let tempdir = copy_directory_to_tempdir(CONTRACTS_DIR.to_string() + "/virtual_workspace");
+    let accounts_json_path = get_accounts_path("tests/data/accounts/accounts.json");
+    let args = vec![
+        "--url",
+        URL,
+        "--accounts-file",
+        accounts_json_path.as_str(),
+        "--account",
+        "user8",
+        "--int-format",
+        "--json",
+        "declare",
+        "--package",
+        "cast_fibonacci",
+        "--contract-name",
+        "FibonacciContract",
+        "--max-fee",
+        "99999999999999999",
+    ];
+
+    let snapbox = Command::new(cargo_bin!("sncast"))
+        .current_dir(tempdir.path())
+        .args(args);
+
+    let output = snapbox.assert().success().get_output().clone();
+    let output = output.stdout.clone();
+    let hash = get_transaction_hash(&output);
+    let receipt = get_transaction_receipt(hash).await;
+    assert!(matches!(receipt, Declare(_)));
+}
+
+#[tokio::test]
+async fn test_worskpaces_package_no_contract() {
+    let tempdir = copy_directory_to_tempdir(CONTRACTS_DIR.to_string() + "/virtual_workspace");
+    let accounts_json_path = get_accounts_path("tests/data/accounts/accounts.json");
+    let args = vec![
+        "--url",
+        URL,
+        "--accounts-file",
+        accounts_json_path.as_str(),
+        "--account",
+        "user8",
+        "--int-format",
+        "declare",
+        "--package",
+        "cast_addition",
+        "--contract-name",
+        "whatever",
+        "--max-fee",
+        "99999999999999999",
+    ];
+
+    let snapbox = Command::new(cargo_bin!("sncast"))
+        .current_dir(tempdir.path())
+        .args(args);
+    snapbox.assert().success().stderr_matches(indoc! {r"
+        ...
+        command: declare
+        error: Failed to find artifacts in starknet_artifacts.json file[..]
```

Comment:
It would be nice to include the contract name in the error message.

Smth like `Failed to find artifacts in starknet_artifacts.json file for {contract_name} contract`